### DISCUSSION TITLE
Option to add questions yourself

### DISCUSSION THREAD

**Discussion by kar1m555** [voidbydefault/EasyApplyBot#37] [General]
we should have an option to add questions ourselves that the code does not know how to deal with. For example, questions that go to the "Unprepared_Questions.cvs" file can be solved by allowing us to adding the answers to these questions ourselves. Let me know if this is a possibility or if i can do it myself to the code we currently have.

**Comment by voidbydefault** [comment]
You can add in `linkedineasyapply.py`, but it's not straightforward; however, it's not impossible either. A bit of reading and understanding code would be helpful. Alternatively, share the most commonly failed questions so I may analyze and compare them with mine to push an update so everyone benefits.

**Comment by sarynn907** [comment]
It's asking if you're over 18 or not and it keeps selecting No. I'm 40, so obviously that's not good lol

**Reply by voidbydefault** [reply]
> It's asking if you're over 18 or not 

Bot isn't designed to handle this question as it's not a very frequently asked question and, frankly, first time to hear a question like this since I started this project. 

Bot defaults to ‘no‘ for whatever it doesn't know.

Edit: if you can handle little bit of coding, go to LinkedIn apply dot py, find "drop-down" questions, there you will see code with keywords, and copy paste one of those, add keywords relevant to 18ish question and set default to yes.

**Comment by sarynn907** [comment]
Right, I had gotten it rather frequently yesterday, to the point where I
had to stop the bot since I'm not even slightly versed in python. I could
send you my  csv file if you'd like.

**Reply by voidbydefault** [reply]
> send you my csv file if you'd like.

I am a finance guy handling this project without formal training in Python. Sorry, can't provide personalized support. Please don't mind.

**Comment by sarynn907** [comment]
I understand, I did watch your videos, what you made is awesome for not
being a career coder. I appreciate you even responding.

**Comment by voidbydefault** [comment]
No worries, happy to help people in finding jobs. To help you particular request, add these lines in `linikedeasyapply.py` go to line like 680 and:

```
                    elif 'above 18' in question_text.lower():  # Check for "above 18" in the question text
                        choice = ""
                        for option in options:
                            if 'yes' in option.lower():  # Select 'yes' option
                                choice = option
                        if choice == "":
                            choice = options[0]  # Default to the first option if 'yes' is not found
                        self.select_dropdown(dropdown_field, choice)


```
Change 'above 18' to keywords you are facing. That's all (besides will) you would need to get going.

**Comment by sarynn907** [comment]
You're amazing!

**Comment by sarynn907** [comment]
I just wanted to thank you, your program helped me land some interviews.
I'm so appreciative.

**Reply by voidbydefault** [reply]
> I just wanted to thank you, your program helped me land some interviews. I'm so appreciative.

Thanks for compliments Good luck mate.
